### ISSUE TITLE
Hyperopt error when no result

### ISSUE THREAD

**Issue by glonlas** [gcarq/freqtrade#337]
## Step 2: Describe your environment

  * Python Version: 3.6.3 (`python -V`)
  * Branch: Develop
  * Last Commit ID: a18019e8e2c339230767d40a1633534482a882b9 (`git log --format="%H" -n 1`)
 
## Step 3: Describe the problem:
Hyperopt throws an error when it is not able to find a result. It happens when we try with very few epochs (like `-e 5`).

### Steps to reproduce:

  1. Execute many times hyperopt with less than 5 epochs: `python3 ./freqtrade/main.py -c config.json hyperopt -e 5`
  2. You will have some hyperopt with a result, until one will fail

### Observed Results:

  * **What happened?** When Hyperopt is too tired with ridiculous attempts with tiny epochs it throws to us the error: `ValueError: attempt to get argmin of an empty sequence`. Because of that our Travis-CI fails time to time.
  * **What did you expect to happen?** We are expecting Hyperopt to return an empty result and be sorry to haven't be able to find something. Not to throw an exception to us.

### Relevant code exceptions or logs:

  ```
Validating configuration ...
.....Traceback (most recent call last):
  File "freqtrade/.env/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 52, in _wrapfunc
    return getattr(obj, method)(*args, **kwds)
AttributeError: 'list' object has no attribute 'argmin'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./freqtrade/main.py", line 472, in <module>
    main()
  File "./freqtrade/main.py", line 412, in main
    args.func(args)
  File "freqtrade/freqtrade/optimize/hyperopt.py", line 245, in start
    best = fmin(fn=optimizer, space=SPACE, algo=tpe.suggest, max_evals=TOTAL_TRIES, trials=trials)
  File "freqtrade/.env/lib/python3.6/site-packages/hyperopt/fmin.py", line 307, in fmin
    return_argmin=return_argmin,
  File "freqtrade/.env/lib/python3.6/site-packages/hyperopt/base.py", line 635, in fmin
    return_argmin=return_argmin)
  File "freqtrade/.env/lib/python3.6/site-packages/hyperopt/fmin.py", line 322, in fmin
    return trials.argmin
  File "freqtrade/.env/lib/python3.6/site-packages/hyperopt/base.py", line 590, in argmin
    best_trial = self.best_trial
  File "freqtrade/.env/lib/python3.6/site-packages/hyperopt/base.py", line 585, in best_trial
    best = np.argmin(losses)
  File "freqtrade/.env/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 1068, in argmin
    return _wrapfunc(a, 'argmin', axis=axis, out=out)
  File "freqtrade/.env/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 62, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "freqtrade/.env/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 42, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
ValueError: attempt to get argmin of an empty sequence
  ```

**Comment by glonlas** [comment]
Will be fixed with the PR #338

**Comment by shusso** [comment]
Good catch @glonlas 👍
